cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java

```diff
@@ -17,53 +17,137 @@
+import io.cdap.cdap.proto.operation.OperationType;
 import io.cdap.http.HttpHandler;
 import io.cdap.http.HttpResponder;
 import io.netty.handler.codec.http.FullHttpRequest;
 import io.netty.handler.codec.http.HttpRequest;
 import io.netty.handler.codec.http.HttpResponseStatus;
-import java.util.ArrayList;
-import java.util.List;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 import javax.ws.rs.PathParam;
 import javax.ws.rs.QueryParam;
 
-/**
- * The {@link HttpHandler} for handling REST calls to operation endpoints.
- */
+/** The {@link HttpHandler} for handling REST calls to operation endpoints. */
 @Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/operations")
 public class OperationHttpHandler extends AbstractAppFabricHttpHandler {
-
+  private final CConfiguration cConf;
   private static final Gson GSON = new Gson();
-  
-  OperationHttpHandler() {
+  private final OperationLifecycleManager operationLifecycleManager;
+  private final int batchSize;
+  public static final String OPERATIONS_LIST_PAGINATED_KEY = "operations";
+
+  @Inject
+  OperationHttpHandler(CConfiguration cConf, OperationLifecycleManager operationLifecycleManager) throws Exception {
+    this.cConf = cConf;
+    this.batchSize = this.cConf.getInt(AppFabric.STREAMING_BATCH_SIZE);
+    this.operationLifecycleManager = operationLifecycleManager;
   }
 
   /**
    * API to fetch all running operations in a namespace.
    *
    * @param namespaceId Namespace to fetch runs from
-   * @param pageToken the token identifier for the current page requested in a paginated
-   *     request
+   * @param pageToken the token identifier for the current page requested in a paginated request
    * @param pageSize the number of application details returned in a paginated request
-   * @param filter optional filters in EBNF grammar. Currently Only one status and one type
-   *     filter is supported with AND expression.
+   * @param filter optional filters in EBNF grammar. Currently Only one status and one type filter
+   *     is supported with AND expression.
    */
   @GET
   @Path("/")
-  public void scanOperations(HttpRequest request, HttpResponder responder,
+  public void scanOperations(
+      HttpRequest request,
+      HttpResponder responder,
       @PathParam("namespace-id") String namespaceId,
       @QueryParam("pageToken") String pageToken,
       @QueryParam("pageSize") Integer pageSize,
-      @QueryParam("filter") String filter) {
-    // TODO(samik, CDAP-20812) fetch the operation runs from store
-    List<OperationRun> runs = new ArrayList<>();
-    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(runs));
+      @QueryParam("filter") String filter)
+      throws BadRequestException, IOException {
+    validateNamespace(namespaceId);
```

Comment:
Add a todo for rbac check with a corresponding jira

Reply:
Done

Reply:
Also, I think these handlers should have a feature flag check.

Reply:
Got it. I have added the check for this feature flag: SOURCE_CONTROL_MANAGEMENT_MULTI_APP